File: 153-FindMinimumInRotatedSortedArray/153-FindMinimumInRotatedSortedArray.py
# Last updated: 1/14/2026, 5:18:16 PM
class Solution:
    def findMin(self, nums: List[int]) -> int:
        l, r = 0, len(nums) - 1

        while l < r:
            mid = (l + r) // 2
            if nums[r] < nums[mid]:
                l = mid + 1
            elif nums[r] > nums[mid]:
                r = mid
            else:
                r -= 1

        
        return nums[l]

File: 153-FindMinimuminRotatedSortedArray/153-FindMinimuminRotatedSortedArray.cpp
// Last updated: 5/15/2026, 10:45:48 AM
1class Solution {
2public:
3    int findMin(vector<int>& nums) {
4        int l = 0, r = nums.size() - 1;
5        while (l < r) {
6            int m = l + (r - l) / 2;
7            if (nums.at(m) > nums.at(r)) {
8                l = m + 1;
9            } else {
10                r = m;
11            }
12        }
13        return nums.at(l);
14    }
15};